AAXtoMP3

```diff
@@ -1,166 +1,325 @@
-                # the other container formats come with their own
-                # tagging mechanisms.
-                id3_version_param=""
-                if test "${extension}" = "mp3"; then
-                   id3_version_param="-id3v2_version 3"
-                fi
-
-                </dev/null ffmpeg -loglevel error -stats -i "${full_file_path}" -i "${cover_path}" -ss "${start%?}" -to "${end}" -map 0:0 -map 1:0 -acodec copy ${id3_version_param} \
-                    -metadata:s:v title="Album cover" -metadata:s:v comment="Cover (Front)" -metadata track="${chapternum}" -metadata title="${chapter_title}" \
-                    "${chapter_file}"
+  fi
+  full_file_path="${output_directory}/${title}.${extension}"
+  bitrate="$(get_bitrate)k"
+  album_artist="$(get_metadata_value album_artist)"
+  album="$(get_metadata_value album)"
+  album_date="$(get_metadata_value date)"
+  copyright="$(get_metadata_value copyright)"
+
+  mkdir -p "${output_directory}"
+
+  # Big long DEBUG output.  Fully describes the settings used for transcoding.  I could probably do this better.
+  # Not this is a long debug command. It's not critical to operation. It's purely for people debugging
+  # and coders wanting to extend the script.
+  debug "$(printf '\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %s\n%-18s: %sn%-18s: %s' title "${title}" auth_code "${auth_code}" mode "${mode}" path "${path}" codec "${codec}" bitrate "${bitrate}" artist "${artist}" album_artist "${album_artist}" album "${album}" album_date "${album_date}" genre "${genre}" copyright "${copyright}" full_file_path "${full_file_path}" metadata_file "${metadata_file}" working_directory "${working_directory}" )"
+
+  # -----
+  # This is the main work horse command.  This is the primary transcoder.
+  # This is the primary transcode. All the heavy lifting is here.
+  </dev/null ffmpeg -loglevel error -stats -activation_bytes "${auth_code}" -i "${path}" -vn -codec:a "${codec}" -ab ${bitrate} -map_metadata -1 -metadata title="${title}" -metadata artist="${artist}" -metadata album_artist="${album_artist}" -metadata album="${album}" -metadata date="${album_date}" -metadata track="1/1" -metadata genre="${genre}" -metadata copyright="${copyright}" "${full_file_path}"
+
+  log "Created ${full_file_path}."
+  # -----
+
+  # Grab the cover art if available.
+  cover_path="${output_directory}/cover.jpg"
+  log "Extracting cover into ${cover_path}..."
+  </dev/null ffmpeg -loglevel error -activation_bytes "${auth_code}" -i "${path}" -an -codec:v copy "${cover_path}"    
+
+  # -----
+  # OK now spit the file if that's what you want.
+  # If we want multiple file we take the big mp3 and split it by chapter.
+  # Not all audio encodings make sense with multiple chapter outputs.  See options section
+  # for more detail
+  if [ "${mode}" == "chaptered" ]; then
```

Comment:
We need to check `${extension} == "mp3`, or `AAXtoMP3 -e:m4b -c` would generate useless m3u file, too.

Reply:
Actually it's not useless.  The playlist is still valid and works with some players.

However there was a couple of missing ; in the options.  Fixing and retesting now.